numpyro/infer/mcmc.py

```diff
@@ -615,6 +621,257 @@ def __init__(self,
         self._algo = 'NUTS'
 
 
+def _get_proposal_loc_and_scale(samples, loc, scale, new_sample):
```

Comment:
can you please add more comments to this method? will be easier to review

Reply:
done; this function is also tested in `test_get_proposal_loc_and_scale`. :)